contracts/facade/factories/ExchangeRateOracleFactory.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity 0.8.19;
+
+import { FIX_ONE, divuu } from "../../libraries/Fixed.sol";
+
+// weird circular inheritance preventing us from using proper IRToken, not worth figuring out
+interface IMinimalRToken {
+    function basketsNeeded() external view returns (uint192);
+
+    function totalSupply() external view returns (uint256);
+}
+
+contract ExchangeRateOracle {
+    error MissingRToken();
+    
+    address public immutable rToken;
+
+    constructor(address _rToken) {
+        // allow address(0)
+        rToken = _rToken;
+    }
+
+    function exchangeRate() public view returns (uint256) {
+        address _rToken = rToken;
+        if (_rToken == address(0)) revert MissingRToken();
+        
+        uint256 supply = IMinimalRToken(_rToken).totalSupply();
+        if (supply == 0) return FIX_ONE;
+
+        return divuu(uint256(IMinimalRToken(_rToken).basketsNeeded()), supply);
+    }
+
+    // basic chainlink interface sufficient for Morpho
+    function latestRoundData()
+        external
+        view
+        returns (
+            uint80 roundId,
+            int256 answer,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        )
+    {
+        return (0, int256(exchangeRate()), 0, 0, 0);
```

Comment:
I don't think `startedAt` and `updatedAt` can be zero here, but not sure if Morpho requires that (could use `block.timestamp` for both?). I'd also set the `roundId` and `answeredInRound` to `1` instead of `0`.